syntax = "proto2";


message Connect {
  optional int32 message_id = 1;
  optional int32 player_id = 2;
}
message GameStarted {
  optional int32 message_id = 1;
}
message GameStartedOk {
  optional int32 message_id = 1;
  optional int32 player_id = 2;
}
message ShootOk {
  optional int32 message_id = 1;
  optional int32 player_id = 2;
}
message GotShotOk {
  optional int32 message_id = 1;
  optional int32 player_id = 2;
}

message PlayerState {
  optional int32 message_id = 1;
  optional int32 player_id = 2;
  optional int32 x = 3;
  optional int32 y = 4;
  optional string direction = 5;
  optional int32 speed = 6;
}

message ShootEvent {
  optional int32 message_id = 1;
  optional int32 projectile_id = 2;
  optional int32 x = 3;
  optional int32 y = 4;
  optional int32 damage = 5;
  optional string direction = 6;
  optional int32 speed = 7;
  optional int32 player_id = 8;
}

message GotShotEvent {
  optional int32 message_id = 1;
  optional int32 projectile_id = 2;
  optional int32 player_id = 8;
}

message GameState {
  optional int32 message_id = 1;
  repeated PlayerState players = 2;
  repeated ShootEvent projectiles = 3;
}
